src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -399,47 +425,61 @@ private void searchForRelations(BibEntry entry, CheckListView<CitationRelationIt
 
         listView.setItems(observableList);
 
+        // TODO: It should not be possible to cancel a search task that is already running for same tab
         if (citingTask != null && !citingTask.isCancelled() && searchType == CitationFetcher.SearchType.CITES) {
```

Comment:
For nullable things we typically use `Option`. (Even though IDEA will warn that `oh no optional is used as a field`.

But still, `Optional` is better

Reply:
`citedByTask != null && !citedByTask.isCancelled()`

Could be rewritten like `citedByTask.map(BackgroundTask::isCancelled).orElse(false)` (could you please double-check if any parameters are needed in `BackgroundTask::`).

@koppor, using Optionals seems to be a little bit more verbose. Should we still use Optional there instead of null?